src/audio.h

```diff
@@ -42,8 +41,29 @@ namespace audio {
     std::bitset<MAX_FLAGS> flags;
   };
 
+  struct audio_ctx_t {
+    // We want to change the sink for the first stream only
+    std::unique_ptr<std::atomic_bool> sink_flag;
+
+    std::unique_ptr<platf::audio_control_t> control;
+
+    bool restore_sink;
+    platf::sink_t sink;
+  };
+
   using buffer_t = util::buffer_t<std::uint8_t>;
   using packet_t = std::pair<void *, buffer_t>;
+  using audio_ctx_ref_t = safe::shared_t<audio::audio_ctx_t>::ptr_t;
+
   void
   capture(safe::mail_t mail, config_t config, void *channel_data);
+
+  /*!
+   * Returns the reference to the audio context.
+   *
+   * @note aside from the configuration purposes, it can be used to extend the
+   *       audio sink lifetime to capture sink earlier and restore it later.
+   */
```

Comment:
I will do this, but this will take some time as the next week is busy for me.

However, I would like to keep the comment block in the headers, at least for the shared declarations in `src/display_device`. The "public" API is usually pretty generic - "verify data, call some function and return result". The complicated stuff is "some function" part which I can then do the commentary for in the source.